src/components/form/HdInputPhone.vue

```diff
@@ -0,0 +1,368 @@
+          countryNode = this.$refs[this.focusedCountry][0]?.nextSibling;
+        }
+        foucsCountryCode();
+      } else if (keyCode === ARROW_UP_KEY_CODE) {
+        if (this.focusedCountry === null) { // If there is no focus yet, focus last option
+          [countryNode] = this.$refs[lastCountryCode];
+        } else if (this.focusedCountry !== firstCountryCode) { // If there is, focus prev option
+          countryNode = this.$refs[this.focusedCountry][0]?.previousSibling;
+        }
+        foucsCountryCode();
+      } else if (isAlphabetKey(keyCode)) {
+        const keyString = event.key.toUpperCase();
+        const firstMatchingCountryCode = this.sortedCountriesList.find(
+          (country) => this.t.COUNTRIES[country.code].charAt(0) === keyString,
+        )?.code;
+        if (firstMatchingCountryCode) {
+          [countryNode] = this.$refs[firstMatchingCountryCode];
+          foucsCountryCode();
+        }
+      } else if (keyCode === ESC_KEY_CODE) {
+        this.toggleDropdown();
+      }
+    },
+    handleInputEvent(value) {
+      this.$emit('input', formatPhoneNumber(this.selectedCountry.dial_code, value));
+    },
+  },
+};
+</script>
+
+<style lang="scss" scoped>
+@import 'homeday-blocks/src/styles/mixins.scss';
+
+$flag-icons-path: '../../../node_modules/flag-icons-svg/svg';
+@import './node_modules/flag-icons-svg/sass/_variables.scss';
+@import './node_modules/flag-icons-svg/sass/flag-icons.scss';
+
+$dropdown-button-width: 74px;
+$dropdown-button-height: 55px;
+
+.phone-input {
+  position: relative;
+  display: flex;
+
+  ::v-deep {
+    .field__main .input {
+      border-top-left-radius: 0 !important;
+    }
+    .field__border {
+      left: -$dropdown-button-width;
+      width: calc(100% + #{$dropdown-button-width});
+    }
+    .field__helper {
+      transform: translateX(-$dropdown-button-width);
+      width: calc(100% + #{$dropdown-button-width});
+    }
+  }
+
+  &__selector {
+    width: $dropdown-button-width;
+    height: $dropdown-button-height;
+    padding: 0;
+    display: flex;
+    align-items: center;
+    justify-content: flex-end;
+    background: $secondary-bg;
+    border: none;
+    border-top-left-radius: $sp-xs;
+    border-right: $sp-xxs solid getShade($quaternary-color, 60);
+
+    &:focus-visible {
+      z-index: 1;
+      border-color: transparent;
+    }
+    &__flag {
+      font-size: 18px;
+    }
+    &__arrow {
+      &--up {
+        transform: rotate(90deg);
+      }
+      &--down {
+        transform: rotate(-90deg);
+      }
+      ::v-deep path {
+        fill: $neutral-gray;
+      }
+    }
+  }
+
+  &__field {
+    flex: 1;
+  }
+
+  &__dropdown {
+    position: absolute;
+    top: $dropdown-button-height + 1;
+    left: 0;
+    z-index: 10;
+    width: 100%;
+    max-height: 264px;
+    overflow: auto;
+    border-radius: $sp-xs;
+    background-color: $white;
+    box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.5);
+    &__option {
+      .option-country {
```

Comment:
Current BEM Structure:

- `phone-input__dropdown`: `<ul>`
    - `phone-input__dropdown__option`: `<li>`
        - `option-country`: `<button>`
            - `option-country_name`: `<span>`, `<p>`, etc.

What do you think about having something like this instead? ⏬ 

- `phone-input__dropdown`: `<ul>` ✅ 
    - `phone-input__dropdown__option`: `<li>` ✅ 
        - `option__country`: `<button>` 🟧 
            - `option__country__name`: `<span>`, `<p>`, etc.  🟧

Furthermore, you can check this exact section of the article to also get more information regarding this topic: https://www.smashingmagazine.com/2016/06/battling-bem-extended-edition-common-problems-and-how-to-avoid-them/#8-how-to-nest-components

Reply:
oh nice! I agree, I will change it